deepcave/runs/converters/amltk.py

```diff
@@ -0,0 +1,216 @@
+    """
+
+    prefix = "AMLTK"
+    _initial_order = 2
+
+    @property
+    def hash(self) -> str:
+        """
+        Hash of the current run.
+
+        If the hash changes, the cache has to be cleared.
+        This ensures that the cache always holds the latest results of the run.
+
+        Returns
+        -------
+        str
+            The hash of the run.
+        """
+        if self.path is None:
+            return ""
+
+        # Use hash of history.csv as id
+        return file_to_hash(self.path / "history.csv")
+
+    @property
+    def latest_change(self) -> Union[float, int]:
+        """
+        Get the timestamp of the latest change.
+
+        Returns
+        -------
+        Union[float, int]
+            The latest change.
+        """
+        if self.path is None:
+            return 0
+
+        return Path(self.path / "history.csv").stat().st_mtime
+
+    @classmethod
+    def from_path(cls, path: Union[Path, str]) -> "AMLTKRun":
+        """
+        Based on working_dir/run_name/*, return a new trials object.
+
+        Parameters
+        ----------
+        path : Union[Path, str]
+            The path to base the trial object on.
+
+        Returns
+        -------
+        The AMLTK run.
+
+        Raises
+        ------
+        RuntimeError
+            Instances are not supported.
+        """
+        path = Path(path)
+
+        # Read configspace
+        from ConfigSpace.read_and_write import json as cs_json
+
+        with open(path / "configspace.json", "rb") as f:
+            json_string = pickle.load(f)
+        configspace = cs_json.read(json_string)
+
+        history = pd.read_csv(path / "history.csv")
+
+        history["budget"] = history["name"].apply(
+            lambda x: float(value) if (value := extract_value(x, "budget")) is not None else None
+        )
+        history["instance"] = history["name"].apply(lambda x: extract_value(x, "instance"))
+
+        # Extract the objectives from the dataframe
+        obj_list = list()
+        for metric_string in history.columns:
+            if metric_string.startswith("metric:"):
+                match = re.match(
+                    r"metric:(\w+) \[(\d+\.\d+), (\d+\.\d+)\] \((\w+)\)", metric_string
+                )
+                assert match is not None
+                metric_name = match.group(1)
+                lower = float(match.group(2))
+                upper = float(match.group(3))
+                maximize = match.group(4) == "maximize"
+
+                obj_list.append(
+                    Objective(
+                        name=metric_name,
+                        lower=lower,
+                        upper=upper,
+                        optimize="upper" if maximize else "lower",
+                    )
+                )
+
+        obj_list.append(Objective("Time"))
+
+        # Let's create a new run object
+        run = AMLTKRun(name=path.stem, configspace=configspace, objectives=obj_list, meta=None)
```

Comment:
Technically anything under `summary:` could be considered user provided `meta=`

Reply:
This is meta information per trial though, right? I did not add it as we currently do not make use of per-trial meta information (only information on the overall run, such as SMAC version, facade etc).